src/api/api.cpp

```diff
@@ -373,12 +469,41 @@ extern "C" struct migraphx_quantize_int8_options;
 struct migraphx_quantize_int8_options
 {
     template <class... Ts>
-    migraphx_quantize_int8_options(Ts&&... xs) : object(std::forward<Ts>(xs)...)
+    migraphx_quantize_int8_options(Ts&&... xs)
+        : object(std::forward<Ts>(xs)...) // NOLINT(readability-redundant-member-init)
     {
     }
     migraphx::quantize_int8_options object;
 };
 
+extern "C" struct migraphx_experimental_custom_op;
+struct migraphx_experimental_custom_op
+{
+    template <class... Ts>
+    migraphx_experimental_custom_op(void* p,
+                                    migraphx_experimental_custom_op_copy c,
+                                    migraphx_experimental_custom_op_delete d,
```

Comment:
should it also have `compute_shape_f`  as one of the args ? or the does it have to set explicitly by calling `set_compute_f` ?

Reply:
Only the copy and delete are passed since those functions are necessary to manage the object. All other functions are set using the set functions like `migraphx_experimental_custom_op_set_compute_shape`.

Set functions are used(instead of structs or parameters) so we can extend the interface without breaking ABI. Furthermore to simplify code gen, all methods must use set methods except those required to manage the objects lifetime(ie copy and delete), rather than start with the initial methods as parameters and use set methods for additional methods added in the future since that requires code gen to support two type of generations.